examples/miscellaneous/miscellaneous_ex12/miscellaneous_ex12.C

```diff
@@ -0,0 +1,750 @@
+// The libMesh Finite Element Library.
+// Copyright (C) 2002-2016 Benjamin S. Kirk, John W. Peterson, Roy H. Stogner
+
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+
+// <h1>Miscellaneous Example 12 - MITC4 Shell Elements</h1>
+// \author Sylvain Vallaghe
+// \date 2016
+//
+// This example implements a variation of MITC4 shell elements.
+// The infamous "pinched cylinder" problem is solved and the solution
+// is compared to analytical values at selected points.
+// The implementation follows very closely the notations of the french
+// reference book on structural analysis with finite elements:
+// Batoz, Jean-Louis, and Gouri Dhatt.
+// "Modelisation des structures par elements finis, Vol. 3: Coques." Hermes, Paris (1992).
+
+// C++ include files that we need
+#include <iostream>
+
+// LibMesh includes
+#include "libmesh/libmesh.h"
+#include "libmesh/mesh.h"
+#include "libmesh/linear_implicit_system.h"
+#include "libmesh/equation_systems.h"
+#include "libmesh/fe.h"
+#include "libmesh/quadrature.h"
+#include "libmesh/node.h"
+#include "libmesh/elem.h"
+#include "libmesh/dof_map.h"
+#include "libmesh/quadrature_gauss.h"
+#include "libmesh/vector_value.h"
+#include "libmesh/tensor_value.h"
+#include "libmesh/dense_matrix.h"
+#include "libmesh/dense_submatrix.h"
+#include "libmesh/dense_vector.h"
+#include "libmesh/dense_subvector.h"
+#include "libmesh/sparse_matrix.h"
+#include "libmesh/numeric_vector.h"
+#include "libmesh/exodusII_io.h"
+#include "libmesh/dirichlet_boundaries.h"
+#include "libmesh/zero_function.h"
+
+//Eigen includes
+#include <Eigen/Dense>
```

Comment:
This (and hence the whole example) needs to be wrapped in ifdef LIBMESH_HAVE_EIGEN